pkg/services/ngalert/models/image.go

```diff
@@ -17,6 +21,46 @@ type Image struct {
 	ExpiresAt time.Time `xorm:"expires_at"`
 }
 
+// ExtendDuration extends the expiration time of the image. It can shorten
+// the duration of the image if d is negative.
+func (i *Image) ExtendDuration(d time.Duration) {
+	i.ExpiresAt = i.ExpiresAt.Add(d)
+}
+
+// HasExpired returns true if the image has expired.
+func (i Image) HasExpired() bool {
+	return time.Now().After(i.ExpiresAt)
+}
+
+// HasFileOnDisk returns true if the image has a path on disk and a file
+// exists at this path. It returns an error if os.Stat returns an error
+// other than os.ErrNotExists.
+func (i Image) HasFileOnDisk() (bool, error) {
+	if !i.HasPath() {
+		return false, nil
+	}
+	if info, err := os.Stat(i.Path); err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return false, nil
+		}
+		return false, err
+	} else if info.IsDir() {
+		return false, fmt.Errorf("%s is a dir", i.Path)
+	} else {
+		return true, nil
+	}
+}
+
+// HasPath returns true if the image has a path on disk.
+func (i Image) HasPath() bool {
+	return i.Path != ""
+}
+
+// HasURL returns true if the image has a URL.
+func (i Image) HasURL() bool {
+	return i.URL != ""
+}
```

Comment:
I understand the motivation for passing receiver by value but I think it is better to be consistent and use either of the approaches for same struct, not both. See Go docs https://go.dev/doc/faq#methods_on_values_or_pointers

Reply:
The thinking here is that it tells the caller "we do not change the image"?

Reply:
I have changed all receivers to `*Image`